From 63f84753f2cc34998e38ee9ae140490c12a017bc Mon Sep 17 00:00:00 2001
From: marius <marius@FreeBSD.org>
Date: Sat, 2 Jun 2012 19:15:21 +0000
Subject: [PATCH 04/12] MFC: r236328

Try to finally get the point in time at which bge_add_sysctls() is called
right; it needs to be called before bge_can_use_msi() but in turn requires
bge_flags to be properly set.

Submitted by:	yongari
---
 sys/dev/bge/if_bge.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sys/dev/bge/if_bge.c b/sys/dev/bge/if_bge.c
index 5e7fe03..bbc770a 100644
--- a/sys/dev/bge/if_bge.c
+++ b/sys/dev/bge/if_bge.c
@@ -2368,7 +2368,6 @@ bge_dma_free(struct bge_softc *sc)
 	if (sc->bge_cdata.bge_tx_mtag)
 		bus_dma_tag_destroy(sc->bge_cdata.bge_tx_mtag);
 
-
 	/* Destroy standard RX ring. */
 	if (sc->bge_cdata.bge_rx_std_ring_map)
 		bus_dmamap_unload(sc->bge_cdata.bge_rx_std_ring_tag,
@@ -2894,8 +2893,6 @@ bge_attach(device_t dev)
 	sc = device_get_softc(dev);
 	sc->bge_dev = dev;
 
-	bge_add_sysctls(sc);
-
 	TASK_INIT(&sc->bge_intr_task, 0, bge_intr_task, sc);
 
 	/*
@@ -3041,6 +3038,9 @@ bge_attach(device_t dev)
 		break;
 	}
 
+	/* Add SYSCTLs, requires the chipset family to be set. */
+	bge_add_sysctls(sc);
+
 	/* Set various PHY bug flags. */
 	if (sc->bge_chipid == BGE_CHIPID_BCM5701_A0 ||
 	    sc->bge_chipid == BGE_CHIPID_BCM5701_B0)
-- 
1.7.9.5

